presto-oracle/src/main/java/com/facebook/presto/plugin/oracle/OracleClient.java

```diff
@@ -0,0 +1,184 @@
+    }
+
+    private String[] getTableTypes()
+    {
+        if (synonymsEnabled) {
+            return new String[] {"TABLE", "VIEW", "SYNONYM"};
+        }
+        return new String[] {"TABLE", "VIEW"};
+    }
+
+    @Override
+    protected ResultSet getTables(Connection connection, Optional<String> schemaName, Optional<String> tableName)
+            throws SQLException
+    {
+        DatabaseMetaData metadata = connection.getMetaData();
+        String escape = metadata.getSearchStringEscape();
+        return metadata.getTables(
+                connection.getCatalog(),
+                escapeNamePattern(schemaName, Optional.of(escape)).orElse(null),
+                escapeNamePattern(tableName, Optional.of(escape)).orElse(null),
+                getTableTypes());
+    }
+    @Override
+    public PreparedStatement getPreparedStatement(Connection connection, String sql)
+            throws SQLException
+    {
+        PreparedStatement statement = connection.prepareStatement(sql);
+        statement.setFetchSize(fetchSize);
+        return statement;
+    }
+
+    @Override
+    protected String generateTemporaryTableName()
+    {
+        return "presto_tmp_" + System.nanoTime();
+    }
+
+    @Override
+    protected void renameTable(JdbcIdentity identity, String catalogName, SchemaTableName oldTable, SchemaTableName newTable)
+    {
+        if (!oldTable.getSchemaName().equalsIgnoreCase(newTable.getSchemaName())) {
+            throw new PrestoException(NOT_SUPPORTED, "Table rename across schemas is not supported in Oracle");
+        }
+
+        String newTableName = newTable.getTableName().toUpperCase(ENGLISH);
+        String oldTableName = oldTable.getTableName().toUpperCase(ENGLISH);
+        String sql = format(
+                "ALTER TABLE %s RENAME TO %s",
+                quoted(catalogName, oldTable.getSchemaName(), oldTableName),
+                quoted(newTableName));
+
+        try (Connection connection = connectionFactory.openConnection(identity)) {
+            execute(connection, sql);
+        }
+        catch (SQLException e) {
+            throw new PrestoException(JDBC_ERROR, e);
+        }
+    }
+
+    //* TODO: Add support for creating schema in JDBC based connector
+    /*
+    public void createSchema(JdbcIdentity identity, String schemaName)
+    {
+        throw new PrestoException(NOT_SUPPORTED, "This connector does not support creating schemas");
+    }
+     */
+
+    @Override
+    public Optional<ReadMapping> toPrestoType(ConnectorSession session, JdbcTypeHandle typeHandle)
+    {
+        int columnSize = typeHandle.getColumnSize();
+
+        switch (typeHandle.getJdbcType()) {
+            case Types.CLOB:
+                return Optional.of(varcharReadMapping(createUnboundedVarcharType()));
+            case Types.SMALLINT:
+                return Optional.of(smallintReadMapping());
+            case Types.FLOAT:
+                if (columnSize == 63) {
```

Comment:
Could you add a comment to denote the meaning of 63?

Reply:
actually this is incorrect, there should be a different handling for type in REAL and DOUBLE/FLOAT